src/hevm/src/EVM/Concrete.hs

```diff
@@ -105,7 +105,7 @@ x0 ^ y0 | y0 < 0    = errorWithoutStackTrace "Negative exponent"
                   | otherwise   = g (x * x) ((y - 1) `shiftR` 1) (x * z)
 
 createAddress :: Addr -> W256 -> Addr
-createAddress a n = num $ keccak $ rlpList [rlpWord160 a, rlpWord256 n]
+createAddress a n = num $ keccak $ rlpList [rlpAddrFull a, rlpWord256 n]
```

Comment:
what is the reason for the change here?

Reply:
`rlpWord160` expresses addresses as numbers, in the sense that `0x0000000000000000000000007265636569766572` gets encoded without leading zeros. This gives the wrong creation address